File: 30daysOFcode_Go/day22.go
package main

import (
    "fmt"
)

type Node struct{
	data int
	left *Node
	right *Node
}

func createNode(data int) *Node{
	node := new(Node)
	node.data = data

	return node
}

func insert(root *Node, data int) *Node{
	if(root == nil){
		return createNode(data)
	} 

	var cur *Node
	if(data <= root.data){
		cur = insert(root.left, data);
		root.left = cur;
	} else{
		cur = insert(root.right, data);
		root.right = cur;
	}

	return root;
}

func getHeight(root *Node) int{
	if(root == nil){
		return 0;
	}
	if(root.left == nil && root.right == nil){
		return 0;
	}
	return 1 + max(getHeight(root.left), getHeight(root.right))
}

func main() {
	var n, data int
	var root *Node

	fmt.Scanf("%d", &n)

	for n > 0{
		fmt.Scanf("%d", &data)
		root = insert(root, data)
		n--
	}

	fmt.Println(getHeight(root));
}

func max(a, b int) int{
	if a > b{
		return a
	}
	return b
}

File: 30daysOFcode_Java/day 22.java
import java.util.*;
import java.io.*;

class Node{
    Node left,right;
    int data;
    Node(int data){
        this.data=data;
        left=right=null;
    }
}

class Solution{
   
	public static int getHeight(Node root){
      //Write your code here
       Node p;
       p=root;
      
      if(p==null)
        return -1;
      else
      {     
            int l = getHeight(p.left);
            int r=getHeight(p.right);
            
            if(l>r)
              return l+1;
           else return r+1;    
      }   
    }

	public static Node insert(Node root,int data){
    if(root==null){
      return new Node(data);
    }
    else{
      Node cur;

      if(data<=root.data){
          cur=insert(root.left,data);
          root.left=cur;
      }
      else{
          cur=insert(root.right,data);
          root.right=cur;
      }
      return root;
    }
  }
  public static void main(String args[]){
    Scanner sc=new Scanner(System.in);
    int T=sc.nextInt();
    Node root=null;
    while(T-->0){
        int data=sc.nextInt();
        root=insert(root,data);
    }
    int height=getHeight(root);
    System.out.println(height);
  }
}    